### PR TITLE
Replace unit type with a form type extension

### LINKED ISSUE #27787: Create NumberWithUnitType and IntegerWithUnitType
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Currently there is only TextWithUnitType.

For the new forms we have to create IntegerWithUnitType and numberWithUnitType

Without this we cannot pass optionresolver with a good template.

It's an issue for the dev core Team https://github.com/PrestaShop/PrestaShop/pull/27608#discussion_r813746914

### Describe the solution you'd like

Create NumberWithUnitType and IntegerWithUnitType

### Alternatives you've considered

_No response_

### Additional context

_No response_

### Do you plan to work on this feature?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by tom-combet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Create unit form type extension to avoid creating unit types for all concerned form types.
| Type?             | refacto
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #27787.
| Related PRs       | Closes #27788.
| How to test?      | Test modified fields to see if same display remains. Some examples:<br>- Advanced Parameters > Administration > Upload Quota<br>- Advanced Parameters > Employees > Options > 'Password change time' field<br><br>Or add 'unit' to any integer or number field.

| Possible impacts? | N/A

### Breaking changes

- `TextWithUnitType` has been deleted because it has been replaced by `UnitTypeExtension`.

### CODE REVIEW

**Comment by tom-combet** [comment]
> Hello @tom-combet!
> 
> This is your first pull request on the PrestaShop project. Thank you, and welcome to this Open Source community!

Hey Preston,

Can you recognize me please? It's not my first pull request actually.

Thank you mate, appreciate.

**Comment by matks** [comment]
> > Hello @tom-combet!
> > This is your first pull request on the PrestaShop project. Thank you, and welcome to this Open Source community!
> 
> Hey Preston,
> 
> Can you recognize me please? It's not my first pull request actually.
> 
> Thank you mate, appreciate.

Prestonbot only acknowledges merged Pull Requests. Until a first PR of you has been merged he will continue to cheer you up.

**Comment by tom-combet** [comment]
Haha ok, good to know. That was just amical btw.

File: src/PrestaShopBundle/Form/Admin/Extension/UnitTypeExtension.php [hunk 1]
```diff
@@ -24,52 +24,48 @@
...
+namespace PrestaShopBundle\Form\Admin\Extension;
+
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
 use Symfony\Component\Form\Extension\Core\Type\NumberType;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\Form\FormView;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
-class TextWithUnitType extends AbstractType
```
Comment (Progi1984) [inline src/PrestaShopBundle/Form/Admin/Extension/UnitTypeExtension.php:35]: Please define the BC Break :)

**Review by Progi1984 — COMMENTED** [review]
A small change for documentation.

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @tom-combet 

Thank your for your PR, I tested it and the refacto seems to works well, 
 
I launched an automated test, and test it on : 

- Advenced parameters > Administration > General (lifetime cookie)
- Advenced parameters > Administration > UploadQuotaType (max size attached files/download files/product image)
- Advenced parameters > Employee > employeeOptionsType ( password change time )
- Shop Parameters > CustomerPreferences > Generale Type ( password reset delay )
- Shop Parameters > ProductPreferences > Generale Type ( Short description limit )
- Shipping > Preferences > HandlingType( free shipping weight)
- Order > AddProductRowType ( price tax excluded , price tax included ) 
- Order > EditProductRowType ( price tax excluded , price tax included ) 
- Product > Combination > BulkCombinationPriceType ( weight )
- Product > Combination > CombinationPriceImpactType ( weight )
- Product > Shipping > DimensionsType ( width, height, depth, weight )

This modification add a scroll option on the right of the number with an integer as you can see 
image

you can scroll down and up with your mouse on the arrow to add or reduce the number.

After 16 digits, you can't use the arrow or the scroll of your mouse to add or reduce the number and the modification won't be save

**Review by AureRita — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @tom-combet & @AureRita

**Pull request merged**